### DISCUSSION TITLE
Silk.NET/docs/opengl/c1/1-hello-window

### DISCUSSION THREAD

**Discussion by giscus** [dotnet/Silk.NET#1339] [Webpage Comments]
# Silk.NET/docs/opengl/c1/1-hello-window

Your one-stop-shop for high performance .NET graphics & compute. A .NET Foundation project.

https://dotnet.github.io/Silk.NET/docs/opengl/c1/1-hello-window.html

**Comment by Perksey** [comment]
Super awesome work with the tutorial @skyebird189!

**Reply by ShannonZ** [reply]
How to use silk,net in AvaloniaUI?

**Reply by Perksey** [reply]
@ShannonZ https://github.com/kekekeks/Avalonia-Silk.NET-Example/tree/master/AvaloniaSilkExample

**Comment by EditSIMS** [comment]
Very intuitive tutorial!   Please fix the "next tutorial" link on the "Wrapping up" Section!

The faulty link:
https://dotnet.github.io/Silk.NET/docs/opengl/c1/2-hello-quad.html

Thanks!

**Comment by ds090525** [comment]
My window was not showing up initially. After playing around a bit I found out I had to use the window options to explicitly set `IsVisible` to true and give the window a reasonable starting position. Here is the code that worked for me:
```csharp
using Silk.NET.Windowing;

namespace Something
{
    internal class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("starting...");

            WindowOptions options = new()
            {
                Size = new(800, 800),
                Title = "Foo",
                IsVisible = true,
                Position = new(50, 50),
            };

            IWindow window = Window.Create(options);

            Console.WriteLine("running...");
            window.Run();

            Console.WriteLine("done");
        }
    }
}
```

**Reply by ds090525** [reply]
Ok, I was tired yesterday. I did not notice that the way how the window options are instantiated is different in the tutorial than in my code. I used `new WindowOptions()` but the tutorial uses `WindowOptions.Default`. This is important because I was later unable to get an OpenGL context from the window without changing to `WindowOptions.Default`.
